doc/tutorials/command-line-options.md

```diff
@@ -63,12 +63,151 @@ into the array.
+args/index = 0
+
+[to]
+args = indexed
+args/index = 1
+
+[more/#]
+args = remaining
+```
+
+If an application `app` with the specification above is called as `./app apple banana cherry date`, then the keys will be
+assigned as follows:
+
+- `from = apple`
+- `to = banana`
+- `more/#0 = cherry`
+- `more/#1 = date`
+
+## Sub-Commands
+
+`elektraGetOpts` also supports sub-commands. For example calling `git -p add` and `git add -p` result in different behavior.
+This is because the options that `git` understands are separate from the options that its sub-command `add` knows. However,
+the option `-p` is understood by both. In `git` it is short for `--paginate` and in `add` it is short for `--patch`.
+
+The first thing you need to know about sub-commands is that they imply `posixly = 1` mode. This means **all** options must
+be given before any non-option arguments (such as parameters or sub-commands). Otherwise, we couldn't distinguish between
+`git -p add` and `git add -p`.
+
+A sub-command is created by specifying `command` on a key. To enable sub-command processing the parent key of the whole
+specification must have `command` set to an empty string. All keys marked with `command` **directly** below another
+key `K` marked with `command` (e.g. the parent key) are sub-commands of `K`. It is an error, if the immediate parent of
+a key `X` marked with `command` is not marked with `command` and `X` is not the parent of the whole specification.
+
+To inform the application about the invoked sub-commands, `elektraGetOpts` sets each `command` key to one of two values:
+
+- All invoked sub-commands `S` will be set to `XYZ`, where `XYZ` is the base name of the key whose sub-command of `S` was invoked.
+- Any other sub-command is set to an empty string.
+
+For example consider `./app add more`: The parent key will be set to `add`, the key for `add` will be set to `more` and
+the key for `more` is set to an empty string, because none of its sub-commands were invoked.
+
+Every key considered by `elektraGetOpts` is assigned either to the root command, or a single sub-command. Specifically,
+each key is assigned to the command of its immediate parent. If sub-commands are used and the immediate parent of an `opt`
+or `args` key has no `command` metadata an error occurs. The value of a key will **only** be set by `elektraGetOpts`, if
+the corresponding sub-command was invoked.
+
+Lastly, it is allowed to have keys with `args` and `command` below the same parent. If a matching sub-command is found among
+the `command` keys, processing will continue there. Otherwise, the `args` keys will be considered. This allows an application
+to implement dynamic commands (like `git` or `kdb`) by using the `args=remaining` array to invoke another application.
+
+If an unknown sub-command is encountered without an `args` key, an error is returned.
+
+All of this is best understood with an example:
+
+```ini
+[kdb]
+command = ""
+
+[kdb/printversion]
+type = boolean
```

Comment:
Some description texts here might be helpful. Ideally, this would be identical to the KeySet used in the example below.